extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -600,4 +621,45 @@ static String getCookieSuffix(String tenantId) {
         return !"Default".equals(tenantId) ? "_" + tenantId : "";
     }
 
+    interface AuthenticationChallegeContext {
+
+        Uni<ChallengeData> runBlocking(Supplier<ChallengeData> function);
+
+    }
+
+    private final AuthenticationChallegeContext blockingChallengeContext = new AuthenticationChallegeContext() {
+        @Override
+        public Uni<ChallengeData> runBlocking(Supplier<ChallengeData> function) {
+            return Uni.createFrom().deferred(new Supplier<Uni<? extends ChallengeData>>() {
+                @Override
+                public Uni<ChallengeData> get() {
+                    if (BlockingOperationControl.isBlockingAllowed()) {
```

Comment:
Why have you done this? You don't need a special interface and implementation for this, just run the code directly inside getChallenge

Reply:
@stuartwdouglas Yeah, I just copied and kept it like that in case an idea of passing that interface as `getChallenge` parameter, similarly to how it is done for the identity provider, will be considered, but indeed, for now it is redundant to I'll simplify

Reply:
@stuartwdouglas it should be better now, have a look please